### PR TITLE
[Feature] Split storage test into separate file

### PR DESCRIPTION
**Pull request by Moonm3n**
Why I'm doing:

Be ut binary size is too large, need to split the test.

What I'm doing:

Split storage test into separate file.

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [x] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [x] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by stdpain — DISMISSED** [review]

**Review by xiangguangyxg — APPROVED** [review]

File: run-be-ut.sh [hunk 1]
```diff
@@ -245,14 +245,18 @@ test_files=`find ${STARROCKS_TEST_BINARY_DIR} -type f -perm -111 -name "*test" \
...
     if [ -x "${GTEST_PARALLEL}" ]; then
         ${GTEST_PARALLEL} ${STARROCKS_TEST_BINARY_DIR}/starrocks_test \
             --gtest_filter=${TEST_NAME} \
             --serialize_test_cases ${GTEST_PARALLEL_OPTIONS}
+        ${GTEST_PARALLEL} ${STARROCKS_TEST_BINARY_DIR}/starrocks_storage_test \
+            --gtest_filter=${TEST_NAME} \
+            --serialize_test_cases ${GTEST_PARALLEL_OPTIONS}
     else
         ${STARROCKS_TEST_BINARY_DIR}/starrocks_test $GTEST_OPTIONS --gtest_filter=${TEST_NAME}
+        ${STARROCKS_TEST_BINARY_DIR}/starrocks_storage_test $GTEST_OPTIONS --gtest_filter=${TEST_NAME}
```
Comment (kevincai) [inline run-be-ut.sh:259]: the starrocks_storage_test will be run again at line 270, it is also in $test_files

File: be/test/CMakeLists.txt [hunk 2]
```diff
@@ -295,129 +411,21 @@ set(EXEC_FILES
...
 add_library(starrocks_test_objs OBJECT ${EXEC_FILES})
 SET_TARGET_PROPERTIES(starrocks_test_objs PROPERTIES COMPILE_FLAGS "-fno-access-control")
 add_executable(starrocks_test test_main.cpp $<TARGET_OBJECTS:starrocks_test_objs>)
 
+add_library(starrocks_storage_test_objs OBJECT ${STORAGE_EXEC_FILES})
+SET_TARGET_PROPERTIES(starrocks_storage_test_objs PROPERTIES COMPILE_FLAGS "-fno-access-control")
+add_executable(starrocks_storage_test test_main.cpp $<TARGET_OBJECTS:starrocks_storage_test_objs>)
+
 TARGET_LINK_LIBRARIES(starrocks_test ${TEST_LINK_LIBS})
+TARGET_LINK_LIBRARIES(starrocks_storage_test ${TEST_LINK_LIBS})
```
Comment (kevincai) [inline be/test/CMakeLists.txt:428]: it is not really storage related testing, maybe we can name it starrocks_test_part1, starrocks_test_part2 ..., and also add comments here to let others know why it is doing like this here.

**Review by Astralidea — DISMISSED** [review]

File: be/test/CMakeLists.txt (hunk 2 above)
Comment (Moonm3n) [reply be/test/CMakeLists.txt]: Done, PTAL.

File: run-be-ut.sh (hunk 1 above)
Comment (Moonm3n) [reply run-be-ut.sh]: Done, PTAL.

File: run-be-ut.sh [hunk 3]
```diff
@@ -238,21 +238,26 @@ fi
 cp -r ${STARROCKS_HOME}/be/test/util/test_data ${STARROCKS_TEST_BINARY_DIR}/util/
 
 test_files=`find ${STARROCKS_TEST_BINARY_DIR} -type f -perm -111 -name "*test" \
-    | grep -v starrocks_test \
+    | grep -v starrocks_test_part1 \
```
Comment (kevincai) [inline run-be-ut.sh:241]: `find -name '*test'` will not match `starrocks_test_part*`

File: be/test/CMakeLists.txt [hunk 4]
```diff
@@ -295,129 +422,33 @@ set(EXEC_FILES
...
+add_executable(starrocks_test_part2 test_main.cpp $<TARGET_OBJECTS:starrocks_test_objs_part2>)
 
-add_library(starrocks_test_objs OBJECT ${EXEC_FILES})
-SET_TARGET_PROPERTIES(starrocks_test_objs PROPERTIES COMPILE_FLAGS "-fno-access-control")
-add_executable(starrocks_test test_main.cpp $<TARGET_OBJECTS:starrocks_test_objs>)
+TARGET_LINK_LIBRARIES(starrocks_test_part1 ${TEST_LINK_LIBS})
+TARGET_LINK_LIBRARIES(starrocks_test_part2 ${TEST_LINK_LIBS})
 
-TARGET_LINK_LIBRARIES(starrocks_test ${TEST_LINK_LIBS})
+add_dependencies(starrocks_test_part2 starrocks_test_part1)
```
Comment (kevincai) [inline be/test/CMakeLists.txt:451]: add comment to explain why adding this dependency.

**Review by kevincai — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Review by fzhedu — APPROVED** [review]

**Pull request merged**